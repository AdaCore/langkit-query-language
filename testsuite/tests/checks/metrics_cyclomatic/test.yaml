driver: 'checker'
rule_name: 'Metrics_Cyclomatic_Complexity'
input_sources: ['proc.adb']
