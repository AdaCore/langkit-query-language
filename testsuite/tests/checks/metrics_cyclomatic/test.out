proc.adb:1:1: rule violation: cyclomatic complexity is too high: 7
1  |  procedure Proc (I : in out Integer; S : String) is   --  FLAG (7)
   | _^
   ||
   || ~~~ 58 other lines ~~~
   ||
60 || end Proc;
   ||_________^

proc.adb:8:4: rule violation: cyclomatic complexity is too high: 6
8  |     task body T is   --  FLAG (6)
   | ____^
   ||
   || ~~~ 9 other lines ~~~
   ||
18 ||    end T;
   ||_________^

proc.adb:21:7: rule violation: cyclomatic complexity is too high: 6
21 |        procedure P1 is  --  FLAG (6)
   | _______^
   ||
   || ~~~ 8 other lines ~~~
   ||
30 ||       end P1;
   ||_____________^

proc.adb:32:7: rule violation: cyclomatic complexity is too high: 7
32 |        entry E1 when True is   --  FLAG (7)
   | _______^
   ||
   || ~~~ 8 other lines ~~~
   ||
41 ||       end E1;
   ||_____________^

